### PR TITLE
feat(query): add new setting enable_dst_hour_fix

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Time conversion handles invalid DST by adding an hour. Accuracy not guaranteed.(disable by default)

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16022)

### CODE REVIEW

File: src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs [hunk 1]
```diff
@@ -371,25 +394,52 @@ where T: AsRef<[u8]>
...
+    times: &mut Vec<u32>,
+    force_timestamp_conversion: bool,
+) -> Result<DateTime<Tz>> {
+    d.and_hms_opt(times[0], times[1], times[2])
+        .map(|naive_datetime| unwrap_local_time(tz, force_timestamp_conversion, &naive_datetime))
+        .transpose()?
+        .ok_or_else(|| ErrorCode::BadBytes(format!("Invalid time provided in times: {:?}", times)))
+}
+
+fn unwrap_local_time(
```
Comment (TCeason) [inline src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs:410]: in src/query/functions/src/scalars/datetime.rs, has a same logic function: `convert_local_time`. But I do not known how to   

unify it in one place.

File: src/query/settings/src/settings_default.rs [hunk 2]
```diff
@@ -692,7 +692,14 @@ impl DefaultSettings {
                 }),
                 ("parse_datetime_ignore_remainder", DefaultSettingValue {
                     value: UserSettingValue::UInt64(1),
-                    desc: "Ignore trailing chars when parse string to datetime(disable by default)",
+                    desc: "Ignore trailing chars when parse string to datetime",
+                    mode: SettingMode::Both,
+                    range: Some(SettingRange::Numeric(0..=1)),
+                }),
+                //parse_datetime_ignore_remainder
+                ("force_timestamp_conversion", DefaultSettingValue {
```
Comment (youngsofun) [inline src/query/settings/src/settings_default.rs:700]: some idea about the name of the setting,  ignore me if wrong:

1.  The name of the setting is too generic. Could you choose a more specific name?
2.  not only for type `time_stamp`? 
3.  I see another setting `parse_datetime_ignore_remainder` , maybe they can share the same naming pattern? maybe we will have other time related settings later.

File: src/query/settings/src/settings_default.rs [hunk 3]
```diff
@@ -692,7 +692,14 @@ impl DefaultSettings {
                 }),
                 ("parse_datetime_ignore_remainder", DefaultSettingValue {
                     value: UserSettingValue::UInt64(1),
-                    desc: "Ignore trailing chars when parse string to datetime(disable by default)",
+                    desc: "Ignore trailing chars when parse string to datetime",
+                    mode: SettingMode::Both,
+                    range: Some(SettingRange::Numeric(0..=1)),
+                }),
+                //parse_datetime_ignore_remainder
```
Comment (youngsofun) [inline src/query/settings/src/settings_default.rs:699]: remove this line?

File: src/query/settings/src/settings_default.rs (hunk 2 above)
Comment (TCeason) [reply src/query/settings/src/settings_default.rs:700]: rename `force_timestamp_conversion` -> `enable_dst_hour_fix` is better? cc @sundy-li

File: src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs [hunk 4]
```diff
@@ -371,25 +394,52 @@ where T: AsRef<[u8]>
...
+                    return match tz.from_local_datetime(&res2) {
+                        MappedLocalTime::Single(t) => Ok(t),
+                        MappedLocalTime::Ambiguous(t1, t2) => {
+                            Err(ErrorCode::BadArguments(format!(
+                                "Ambiguous Local Time: The local time is ambiguous {}, {} with timezone {}",
+                                t1, t2, tz
+                            )))
+                        }
+                        MappedLocalTime::None => Err(ErrorCode::BadArguments(format!(
+                            "Local Time Error: The local time {}, {} can not map to a single unique result with timezone {}",
```
Comment (youngsofun) [inline src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs:433]: add contest saying the res2 is added by a hour?

Comment (TCeason) [reply src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs:433]: It's ok. Only after add a hour will return two ts when unwrap none

File: src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs (hunk 1 above)
Comment (youngsofun) [reply src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs:410]: databend-common-functions already deps on databend-common-io？why not pub this fn and call it one from datetime.rs

**Review by youngsofun — APPROVED** [review]

**Review by sundy-li — APPROVED** [review]

**Comment by TCeason** [comment]
cc @wubx dst fix in this pr. Please merge this pr cc @BohuTANG

**Comment by TCeason** [comment]
cc @soyeric128  Doc need

**Pull request merged**